File: HoneyStoreUI/src/app/models/wish.ts
import { Product } from './product';

export class Wish {
    id: number | undefined;
    userId: number | undefined;
    quantity: number;
    createdOn: Date | undefined;
    productId: number;
    product: Product;

    constructor(productId: number = 1, userId: number | undefined = 1, quantity: number = 1) {
        this.id = 0;
        this.productId = productId;
        this.product = new Product();
        this.userId = userId;   
        this.quantity = quantity;
        this.createdOn = new Date(Date.now());
    }
}

File: HoneyStore.DataAccess/Entities/Wish.cs
using HoneyStore.DataAccess.Identity;
using HoneyStore.DataAccess.Interfaces;

namespace HoneyStore.DataAccess.Entities
{
    public class Wish : IIdentifier
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public User User { get; set; }

        public int ProductId { get; set; }

        public Product Product { get; set; }

        public int Quantity { get; set; }

        public DateTime CreatedOn { get; set; }
    }
}
